main_test.go

```diff
@@ -0,0 +1,291 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	testConsent "identity_platform_login_ui/ory_mocking/Consent"
+	testErrors "identity_platform_login_ui/ory_mocking/Errors"
+	testLoginUpdate "identity_platform_login_ui/ory_mocking/Login"
+	testLoginBrowser "identity_platform_login_ui/ory_mocking/LoginBrowser"
+	testServers "identity_platform_login_ui/ory_mocking/Testservers"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// --------------------------------------------
+// TESTING WITH CORRECT SERVERS
+// --------------------------------------------
+func TestHandleCreateFlowWithoutCookie(t *testing.T) {
+	//init clients
+	serverClose := testServers.CreateTestServers()
+	defer serverClose()
+
+	//create request and response objects
+	req := httptest.NewRequest(http.MethodGet, "/api/kratos/self-service/login/browser?aal=aal1&login_challenge=&refresh=false&return_to=http://test.test", nil)
+	req.Header.Set("Content-Type", "application/json")
+	w := httptest.NewRecorder()
+
+	//test and evaluate test
+	handleCreateFlow(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	lbr := new(testLoginBrowser.LoginBrowserResponse)
+	if err := json.Unmarshal(data, lbr); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if lbr.Id != "test_id" {
+		t.Errorf("expected test_id, got %v", string(data))
+	}
+}
+
+func TestHandleCreateFlowWithCookie(t *testing.T) {
+	//init clients
+	serverClose := testServers.CreateTestServers()
+	defer serverClose()
+
+	//create request and response objects
+	req := httptest.NewRequest(http.MethodPut, "/api/kratos/self-service/login/browser?aal=aal1&login_challenge=test_challange&refresh=false&return_to=http://test.test", nil)
+	req.Header.Set("Content-Type", "application/json")
+	cookie := &http.Cookie{
+		Name:   "ory_kratos_session",
+		Value:  "test-token",
+		MaxAge: 300,
+	}
+	req.AddCookie(cookie)
+	w := httptest.NewRecorder()
+
+	//test and evaluate test
+	handleCreateFlow(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	if res.StatusCode != 200 {
+		t.Errorf("expected StatusCode to be 200 got %v", res.StatusCode)
+	}
+	requestLoginResponse := new(testLoginBrowser.OAuth2RequestLoginResponse)
+	if err := json.Unmarshal(data, requestLoginResponse); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
```

Comment:
This is repeated in too many places, perhaps we could move this logic to a `UnamarshalJsonBody` (or something like that) function.

Reply:
I rather not. It's true that it's repeated a lot, however:
- If it's ok with you I rather stick to unmarshaling to the type instead of to some map[string]interface{} construct. I think trying with some reflection magic to send in a type for instantiating would make the tests more complicated than they need to be.
- also you don't save much lines at all. Most of these highlighted lines are actually error handling which you would still handle in the tests, because convention (and my preference is) is that functions don't handle errors just send them back to the caller.